features/bicycle/access.feature

```diff
@@ -182,3 +182,9 @@ Feature: Bike - Access tags on ways
             | bridleway |            | yes  |         | x     |
             | bridleway | designated |      |         |       |
             | bridleway |            |      |         |       |
+
+    Scenario: Bike - Tram with oneway when access is implicit
+        Then routability should be
+            | highway     | railway | oneway | forw | backw |
+            | residential | tram    | yes    |      | x     |
```

Comment:
Here is pretty hidden bug in c++ code that was solved by #2399 

* lua bicycle profile returns `result.forward_mode = mode.train (5), result.forward_speed = -1, result.backward_mode = mode.pushing_bike (6), result.backward_speed = 6`

* [the check](https://github.com/Project-OSRM/osrm-backend/blob/445225bf6dc719ef4f91fe6ab7e6fa2f4941fef7/src/extractor/extractor_callbacks.cpp#L121-L122) fails and `forward_weight_data` is default constructed with `forward_weight_data = {0.0, WeightType::INVALID}`

* [the next check](https://github.com/Project-OSRM/osrm-backend/blob/445225bf6dc719ef4f91fe6ab7e6fa2f4941fef7/src/extractor/extractor_callbacks.cpp#L135-L136) pass, because `forward_weight_data = {6, WeightType::SPEED}`

* [here](https://github.com/Project-OSRM/osrm-backend/blob/445225bf6dc719ef4f91fe6ab7e6fa2f4941fef7/src/extractor/extractor_callbacks.cpp#L291-L297) `split_edge = false` because `parsed_way.forward_speed = -1`

* [here](https://github.com/Project-OSRM/osrm-backend/blob/445225bf6dc719ef4f91fe6ab7e6fa2f4941fef7/src/extractor/extractor_callbacks.cpp#L312) `is_opposite_way = false` because `parsed_way.forward_travel_mode` is `TRAVEL_MODE_TRAIN`

* [here](https://github.com/Project-OSRM/osrm-backend/blob/445225bf6dc719ef4f91fe6ab7e6fa2f4941fef7/src/extractor/extractor_callbacks.cpp#L346-L347) `forward_only` is false because neither `split_edge` nor `TRAVEL_MODE_INACCESSIBLE == parsed_way.backward_travel_mode` are true

* [adding edges](https://github.com/Project-OSRM/osrm-backend/blob/445225bf6dc719ef4f91fe6ab7e6fa2f4941fef7/src/extractor/extractor_callbacks.cpp#L353-L365)
```
osm_source_id = 1, osm_target_id= 2, weight_data = {0.0, WeightType::INVALID}, forward = 1, backward= 1, is_split = 0, travel_mode = 5
osm_source_id = 2, osm_target_id= 3, weight_data = {0.0, WeightType::INVALID}, forward = 1, backward= 1, is_split = 0, travel_mode = 5
osm_source_id = 3, osm_target_id= 4, weight_data = {0.0, WeightType::INVALID}, forward = 1, backward= 1, is_split = 0, travel_mode = 5
osm_source_id = 4, osm_target_id= 5, weight_data = {0.0, WeightType::INVALID}, forward = 1, backward= 1, is_split = 0, travel_mode = 5
```

* now because of [**bug here**](https://github.com/Project-OSRM/osrm-backend/blob/445225bf6dc719ef4f91fe6ab7e6fa2f4941fef7/src/extractor/extraction_containers.cpp#L447-L448) the return value for `data={0.0, WeightType::INVALID}` is -1

* now -1 weight is corrected [here](https://github.com/Project-OSRM/osrm-backend/blob/445225bf6dc719ef4f91fe6ab7e6fa2f4941fef7/src/extractor/extraction_containers.cpp#L454)
to 1

As a result 2 compressed edges with weight 4 and mode `TRAVEL_MODE_TRAIN` that is not the same as desired by profile and allows to drive in both directions.